site/source/tutorials/TypeScript_and_Dojo_2.md

```diff
@@ -0,0 +1,60 @@
+- It supports state of the art JavaScript - TypeScript code is converted to JavaScript by a compiler before it is run. In addition to simply converting the code the compiler is also able to allow advanced JavaScript features to be used, such as [asynchronous functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) and [decorators](http://www.typescriptlang.org/docs/handbook/decorators.html).
+
+The Dojo 2 project has taken advantage of each of these to achieve its goals.
+
+## Dojo 2's goals
+
+Every framework makes design decisions based on the project's goals. The best framework for a given project is often the one whose design goals most closely match the needs of the project.
+
+Dojo 2 was designed to address the needs of large-scale, enterprise applications. To meet this single, overarching target, several smaller goals need to be achieved.
+
+- Dojo 2 needs to ensure that applications can be developed quickly, scaled smoothly, and maintained easily.
+- It must simplify the creation of data-centric applications that are prevalent among many enterprises suites of internal applications.
+- It must provide out-of-the-box support for features that are important to enterprises such as [accessibility (a11y)](https://en.wikipedia.org/wiki/Computer_accessibility), [internationalization (i18n)](https://en.wikipedia.org/wiki/Internationalization_and_localization), and security.
+
+## How TypeScript is helping Dojo 2 achieve its goals
+
+TypeScript offers many features that have enabled Dojo 2 to realize its objectives. Three of the primary features are: the type system, interfaces, and the compiler.
+
+### The type system
+
+On its own, JavaScript is a [dynamic](https://en.wikipedia.org/wiki/Dynamic_programming_language), [untyped](https://en.wikipedia.org/wiki/Programming_language#Type_system) language. While these features make JavaScript very easy to start working with, they can quickly become points of concern. Developers often spend a great deal of time reviewing existing code to be sure that they are using it properly. This can slow down the development process since spending time reading existing code takes away from the time available for writing new features.
+
+TypeScript includes an optional type system that addresses these issues. By taking advantage of type information, tools can increase a developer's productivity by letting them know what data types are expected within the application. Features like code completion, refactoring support, and static analysis help to ensure that the developer is using the existing aspects of the application correctly, allowing the developer to spend more time writing code and less time reading. Additionally, since the type system provides design-time warnings about incompatible types, type assertion tests can be eliminated from the application's test suite.
```

Comment:
I would probably say something before tooling... that it makes the documentation more people and the source code easier to understand, etc.